build.gradle

```diff
@@ -575,3 +575,35 @@ gradle.projectsEvaluated {
     }
   }
 }
+
+if (System.properties.get("build.compare") != null) {
+  apply plugin: 'compare-gradle-builds'
+  compareGradleBuilds {
+    ext.referenceProject = System.properties.get("build.compare")
+    doFirst {
+      if (file(referenceProject).exists() == false) {
+        throw new GradleException(
+                "Use git worktree to check out a version to compare against to ../elasticsearch_build_reference"
+        )
+      }
+    }
+    sourceBuild {
+      gradleVersion = "4.7" // does not default to gradle weapper of project dir, but current version
+      projectDir = referenceProject
+      tasks = ["clean", "assemble"]
+      arguments = ["-Dbuild.compare_friendly=true"]
+    }
+    targetBuild {
+      tasks = ["clean", "assemble"]
+      // use -Dorg.gradle.java.home= to alter jdk versions
+      arguments = ["-Dbuild.compare_friendly=true"]
+    }
+  }
+}
+
+// make sure integration tests run after unit tests
+allprojects {
+  if (tasks.findByPath('test') != null && tasks.findByPath('integTest') != null) {
```

Comment:
Can you put this together in the block around line 319, which also deals with test ordering? What causes this to be a requirement for upgrading to 4.8? Also, separately, I think we should look to make most of these mustRunAfter into shouldRunAfter to allow parallelization.

Reply:
++ on souldRunAfter. It's not a necessity for Gradle 4.8  but I did find it to be useful when debugging the many problems we had with  Gradle 4.8.